src/iframe/storage/plasmo-storage/plasmo-storage.mock.ts

```diff
@@ -0,0 +1,275 @@
+import { Storage as PlasmoStorage } from "@plasmohq/storage";
+import {
+  EnhancedStorage,
+  type ItemStorageOptions,
+  StorageManager
+} from "../unpartitioned-storage/unpartitioned-storage";
+
+export interface StorageMockInterface extends PlasmoStorage {
+  setItem<T>(
+    key: string,
+    value: T,
+    options?: ItemStorageOptions
+  ): Promise<void>;
+  getItem<T>(key: string): Promise<T | undefined>;
+  removeItem(key: string): Promise<void>;
+  getItems<T>(keys: string[]): Promise<Record<string, T | undefined>>;
+  removeItems(keys: string[]): Promise<void>;
+  setItems(
+    items: Record<string, any>,
+    options?: ItemStorageOptions
+  ): Promise<void>;
+  setCache<T>(key: string, value: T, maxAge?: number): Promise<void>;
+  setPreference<T>(key: string, value: T): Promise<void>;
+  setTemporary<T>(key: string, value: T, maxAge?: number): Promise<void>;
+  setCritical<T>(key: string, value: T): Promise<void>;
+  evictIfNeeded(bytesNeeded?: number, aggressive?: boolean): Promise<boolean>;
+  getUsageInfo(forceRefresh?: boolean): {
+    bytes: number;
+    percentage: number;
+    items: number;
+    available: number;
+  };
+  ensureSpace(
+    bytesNeeded: number,
+    options?: { skipCheck?: boolean; forceEviction?: boolean }
+  ): Promise<boolean>;
+  setPrioritizedItem<T>(
+    key: string,
+    value: T,
+    priority: keyof typeof StorageManager.PRIORITY_LEVELS,
+    expiresIn?: number
+  ): Promise<void>;
+  requestStorageAccess(): Promise<void>;
+  requestAccessOnUserInteraction(): Promise<void>;
+  hasAvailableSpace(bytesNeeded: number): Promise<boolean>;
+  getRaw(key: string): Promise<string | null>;
+  setRaw(key: string, value: string): Promise<void>;
+  keys(): Promise<string[]>;
+}
+
+export class StorageMock extends PlasmoStorage implements StorageMockInterface {
+  private storage: EnhancedStorage;
+
+  constructor() {
+    super({ area: "session" });
+    this.storage = new EnhancedStorage({ area: "session" });
+
+    // This browser doesn't support the Storage Access API
+    // so let's just hope we have access!
+    if (!document.hasStorageAccess) return;
```

Comment:
If we come across this (or if users reject the unpartitioned state access request, maybe we should just use the regular `localStorage` but show an alert banner on the app letting them know their wallets will not be available on other sites automatically (without them manually importing or recovering them).